kpops/config/__init__.py

```diff
@@ -20,6 +22,28 @@
 ENV_PREFIX = "KPOPS_"
 
 
+class StrimziTopicConfig(BaseSettings):
+    """Configuration for Strimzi Kafka Topics."""
+
+    resource_label: tuple[str, str] = Field(
+        description="The label to identify the KafkaTopic resources managed by the Topic Operator. This does not have to be the name of the Kafka cluster. It can be the label assigned to the KafkaTopic resource. If you deploy more than one Topic Operator, the labels must be unique for each. That is, the operators cannot manage the same resources."
+    )
+
+    @pydantic.field_validator("resource_label", mode="before")
+    @classmethod
+    def deserialize_topics(cls, resource_label: Any) -> tuple[str, str]:
+        match resource_label:
+            case str(value) if "=" in value:
+                key, value = value.split("=")
+                return key, value
+            case dict(value) if len(value) == 1:
+                key, value = next(iter(value.items()))
+                return key, value
+            case _:
+                msg = "'resource_label' should be defined either like 'foo=bar' or as a valid dictionary."
```

Comment:
why not only allow dict then? Or define the required labels in the Pydantic model directly?

Reply:
The configured `STRIMZI_RESOURCE_LABELS` on the topic operator can have any arbitrary key-value:
```
- bakdata.com/cluster=dev-cluster
- kpops.io/kafka-cluster=test-cluster
- strimizi-cluster=k8s-cluster
- foo=bar
```
Each topic operator must set a unique `<key>=<value>` pair. The `KafkaTopic` CRD must have the same label so that the topic operator can find the CRD and handle (create) the topic.